searx/engines/qwant.py

```diff
@@ -238,19 +243,43 @@ def response(resp):
     return results
 
 
-# get supported languages from their site
 def _fetch_supported_languages(resp):
-    # list of regions is embedded in page as a js object
-    response_text = resp.text
-    response_text = response_text[response_text.find('INITIAL_PROPS') :]
-    response_text = response_text[response_text.find('{') : response_text.find('</script>')]
-
-    regions_json = loads(response_text)
-
-    supported_languages = []
-    for country, langs in regions_json['locales'].items():
-        for lang in langs['langs']:
-            lang_code = "{lang}-{country}".format(lang=lang, country=country)
-            supported_languages.append(lang_code)
+
+    text = resp.text
+    text = text[text.find('INITIAL_PROPS') :]
+    text = text[text.find('{') : text.find('</script>')]
+
+    q_initial_props = loads(text)
+    q_locales = q_initial_props.get('locales')
+    q_valid_locales = []
+
+    for country, v in q_locales.items():
+        for lang in v['langs']:
+            _locale = "{lang}_{country}".format(lang=lang, country=country)
+
+            if qwant_categ == 'news' and _locale.lower() not in qwant_news_locales:
+                # qwant-news does not support all locales from qwant-web:
+                continue
```

Comment:
> But I can't reproduce what you have mentioned should result in an API error. Searching !qw time :zh-TW using the current master branch does not result in an API error. Could further explain this please?

@tiekoetter  I fixed this: its only the qwant-news engine that do not supports some qwant locales / with the fix from above we can see that qwant-news support 24 languages (aka locales) while other qwant engines support 39 locales:

```
$ ./manage pyenv.cmd ./searxng_extra/update/update_languages.py
...
fetched 39 languages from engine qwant
fetched 39 languages from engine qwant images
fetched 24 languages from engine qwant news
fetched 39 languages from engine qwant videos
```